classes/services/PKPFileService.inc.php

```diff
@@ -85,9 +85,25 @@ public function add($from, $to) {
 		if (is_resource($stream)) {
 			fclose($stream);
 		}
+		$mimetype = $this->fs->getMimetype($to);
+
+		// Check and override ambiguous mime types based on file extension
+		$pathinfo = pathinfo($to);
+		if (array_key_exists('extension', $pathinfo)) {
+			$extension = $pathinfo['extension'];
+			$checkAmbiguous = strtolower($extension . ':' . $mimetype);
+			if (array_key_exists($checkAmbiguous, $extensionsMap = \PKPString::getAmbiguousExtensionsMap())) {
+				$mimetype = $extensionsMap[$checkAmbiguous];
+			}
+		}
+
+		if (array_key_exists($mimetype, $mimetypesMap = \PKPString::getAmbiguousExtensionsMap())) {
+			$mimetype = $mimetypesMap[$mimetype];
+		}
```

Comment:
Aren't the lines 100-102 duplication lines 95-97? Either that or you're re-using `$mimetype` as a variable name so it will already be overwritten...

Reply:
Oops, forgot to remove that. One sec...